includes/tracks/class-wc-tracks-footer-pixel.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Send Tracks events on behalf of a user using pixel images in page footer.
+ *
+ * @package WooCommerce\Tracks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Tracks_Footer_Pixel class.
+ */
+class WC_Tracks_Footer_Pixel {
+	/**
+	 * Singleton instance.
+	 *
+	 * @var WC_Tracks_Footer_Pixel
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Events to send to Tracks.
+	 *
+	 * @var array
+	 */
+	protected $events = array();
+
+	/**
+	 * Instantiate the singleton.
+	 *
+	 * @return WC_Tracks_Footer_Pixel
+	 */
+	public static function instance() {
```

Comment:
The changes in https://github.com/woocommerce/woocommerce/pull/22930 would need to be reflected in this file too, yeah?

Reply:
Both `WC_Tracks_Client` and `WC_Tracks_Footer_Pixel` are called statically from `WC_Tracks_Event` so this change isn't quite as straightforward. Are we OK with leaving the singleton for this case?